prow/git/v2/client_factory.go

```diff
@@ -129,7 +136,11 @@ func NewClientFactory(opts ...ClientFactoryOpt) (ClientFactory, error) {
 		return nil, err
 	}
 	var remotes RemoteResolverFactory
-	if o.UseSSH != nil && *o.UseSSH {
+	if o.CloneURI != nil {
```

Comment:
Would expect something more strict than `CloneURI != nil -> Using cloneURIResolverFactory`, just like `o.UseSSH` below.

Reply:
Discussed offline. Added support for PublisherRemote for CloneURIResolverFactory so adding CloneURI won't cause surprising behavior